From f18efe239cb4bbfd6996f753ae694f81041d8d43 Mon Sep 17 00:00:00 2001
From: Vernon Mauery <vernon.mauery@linux.intel.com>
Date: Fri, 14 Feb 2020 13:13:06 -0800
Subject: [PATCH] Fix 'Get System GUID' to use settings UUID

The upstream Get System GUID command looks first for a BMC interface
and then assumes that the UUID interface is next to that. But that is
not the case on Intel systems where the system GUID is found in the
settings daemon.

Change-Id: I924bd05e0a546f2b30288c1faf72157296ab6579
Signed-off-by: Vernon Mauery <vernon.mauery@linux.intel.com>
---
 apphandler.cpp | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/apphandler.cpp b/apphandler.cpp
index 90818a9..dcf2c86 100644
--- a/apphandler.cpp
+++ b/apphandler.cpp
@@ -788,8 +788,6 @@ auto ipmiAppGetBtCapabilities()
 
 auto ipmiAppGetSystemGuid() -> ipmi::RspType<std::array<uint8_t, 16>>
 {
-    static constexpr auto bmcInterface =
-        "xyz.openbmc_project.Inventory.Item.Bmc";
     static constexpr auto uuidInterface = "xyz.openbmc_project.Common.UUID";
     static constexpr auto uuidProperty = "UUID";
 
@@ -798,7 +796,7 @@ auto ipmiAppGetSystemGuid() -> ipmi::RspType<std::array<uint8_t, 16>>
     {
         // Get the Inventory object implementing BMC interface
         auto busPtr = getSdBus();
-        auto objectInfo = ipmi::getDbusObject(*busPtr, bmcInterface);
+        auto objectInfo = ipmi::getDbusObject(*busPtr, uuidInterface);
 
         // Read UUID property value from bmcObject
         // UUID is in RFC4122 format Ex: 61a39523-78f2-11e5-9862-e6402cfc3223
-- 
2.17.1

